frontend/src/reducers/index.jsx

```diff
@@ -48,11 +48,16 @@ const resultsReducer = (state = {}, action) => {
     case ActionTypes.RESULT_LIST_SUCCESS:
       if (action.response && action.response.results) {
         const resultList = action.response.results;
+        const resultIds = resultList.map((result) => result.id);
+        let modified = Object.keys(state).length !== resultIds.length;
         const results = {};
         resultList.forEach((result) => {
-          results[result.id] = result;
+          const oldResult = state[result.id] || {};
+          const resultModified = oldResult.logModifiedAt !== result.logModifiedAt;
+          results[result.id] = resultModified ? result : oldResult;
+          modified = modified || resultModified;
         });
-        return results;
+        return modified ? results : state;
```

Comment:
Do we need to return new state only when `logModifiedAt` has changed?
For example, what if `maxLogCount` changes.

Reply:
I missed this case.
Thanks.